compiler/AST/expr.cpp

```diff
@@ -5479,7 +5487,7 @@ GenRet CallExpr::codegenPrimitive() {
       Type* dst = get(1)->typeInfo();
       Type* src = get(2)->typeInfo();
 
-      if (dst == src) {
+      if (dst == src && !(is_int_type(dst) || is_uint_type(dst)) ) {
```

Comment:
This change worries me a little bit. Is the problem actually that Chapel doesn't always have the complete picture in it's type system? If that's the case, should this be checking for extern types? Why should int/uint receive special treatment here?